metricflow/model/semantics/linkable_spec_resolver.py

```diff
@@ -627,13 +669,24 @@ def _get_metric_time_elements(self, measure_reference: MeasureReference) -> Link
         )
 
         # For each of the possible time granularities, create a LinkableDimension for each one.
-        return LinkableElementSet(
-            path_key_to_linkable_dimensions={
-                ElementPathKey(
+
+        path_key_to_linkable_dimensions: Dict[ElementPathKey, List[LinkableDimension]] = defaultdict(list)
+        for time_granularity in possible_metric_time_granularities:
+            possible_date_parts: Sequence[Optional[DatePart]] = (
+                # No date part, just the metric time at a different grain.
+                (None,)
+                # date part of a metric time at a different grain.
+                + tuple(date_part for date_part in DatePart if time_granularity.to_int() <= date_part.to_int())
```

Comment:
I see this granularity/date_part comparison all over the place and it's confusing every time. Maybe we should add a method to DatePart that does this and document the mechanics of the comparison there.